### DISCUSSION TITLE
[v5] Make sure to set "null" as the default value for foreign keys

### DISCUSSION THREAD

**Discussion by e-e** [adonisjs/core#2433] [Help] [answered]
Full error message:
```
Exception: Cannot preload "gender", value of "User.genderId" is undefined. Make sure to set "null" as the default value for foreign keys
```

I currently have two tables: `users` and `genders`. A user has a gender, which looks like:

users | 
------------ | 
id |
gender_id | 

and

genders | 
------------ | 
id |
label | 

A user record can initially be created before setting the gender, and I want to make sure I don't get any errors and instead return null if trying to load the gender (`await user.load('gender')`), since a null gender is an acceptable state sometimes.

My user model is set up with the relation like so:

```ts
class User extends BaseModel {
  @column({isPrimary: true})
  public id: number;

  @column()
  public genderId: number;

  @belongsTo(() => Gender)
  public gender: BelongsTo<typeof Gender>;
}
```

I've tried doing:

```
@column()
public genderId: number | null = null;
```

but got the same error.

Thanks in advance

**Comment by aldinp16** [comment, marked as answer]
set in user migrations to, maybe help
```ts
      table
        .integer('gender_id')
        .references('id')
        .inTable('genders')
        .defaultTo(null)
        .onDelete('SET NULL')
```

**Reply by e-e** [reply]
Unless I'm mistaken I'm pretty sure the default default value for columns is already `null`

**Reply by aldinp16** [reply]
i try produce the error in REPL. It seems error occurs when after create user and call ```preload()``` without call ```refresh()``` method. try to call ```user.refresh()``` before call ```user.preload('gender')``` after save model instance.

**Reply by e-e** [reply]
Thank you, that was the problem, although it wasn't happening where I thought it was. I was calling `await user.load("gender")` in my test after creating a test user, and I needed to reload the model before doing so.

**Comment by AhmadPirzargar** [comment]
I set directly on the model value to null and this work for me :)

`@column({ columnName: 'shopId' })
  public shopId: number | null = null`
